cpp/src/arrow/compute/function.h

```diff
@@ -225,6 +233,15 @@ class ARROW_EXPORT Function {
   /// required by the kernel.
   virtual Result<const Kernel*> DispatchBest(std::vector<TypeHolder>* values) const;
 
+  /// \brief Get a function executor with a best-matching kernel
+  virtual Result<std::shared_ptr<FunctionExecutor>> BestExecutor(
```

Comment:
Naming-wise, a method which does something non-trivial (as opposed to a simple accessor) should probably be described with a verb, so e.g. `GetBestExecutor`.